external/openglcts/modules/gl/gl4cEnhancedLayoutsTests.cpp

```diff
@@ -27698,6 +27702,838 @@ void XFBCaptureUnsizedArrayTest::testInit()
+
+	/* */
+	return source;
+}
+
+/** Get name of test case
+ *
+ * @param test_case_index Index of test case
+ *
+ * @return Name of tested stage
+ **/
+std::string XFBExplicitLocationStructTest::getTestCaseName(glw::GLuint test_case_index)
+{
+	std::stringstream stream;
+	const testCase&   test_case = m_test_cases[test_case_index];
+
+	stream << "Struct: { ";
+
+	for (const testType& type : test_case.m_types)
+	{
+		stream << type.m_type.GetGLSLTypeName() << "@" << type.m_array_size << ", ";
+	}
+
+	stream << "}, stage: " << Utils::Shader::GetStageName(test_case.m_stage);
+
+	return stream.str();
+}
+
+/** Returns number of test cases
+ *
+ * @return TEST_MAX
+ **/
+glw::GLuint XFBExplicitLocationStructTest::getTestCaseNumber()
+{
+	return static_cast<GLuint>(m_test_cases.size());
+}
+
+/** Prepare all test cases
+ *
+ **/
+void XFBExplicitLocationStructTest::testInit()
+{
+	for (GLuint stage = 0; stage < Utils::Shader::STAGE_MAX; ++stage)
+	{
+		if ((Utils::Shader::COMPUTE == stage) || (Utils::Shader::FRAGMENT == stage) ||
+			(Utils::Shader::TESS_CTRL == stage))
+		{
+			continue;
+		}
+
+		const GLuint n_types = getTypesNumber();
+
+		for (GLuint i = 0; i < n_types; ++i)
+		{
+			const Utils::Type& type = getType(i);
+
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_float, 0}, {type, 0} }, false });
+		}
+
+		for (bool is_nested_struct : {false, true})
+		{
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_double, 0}, {Utils::Type::dvec2, 0}, {Utils::Type::dmat3, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_double, 0}, {Utils::Type::vec3, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::dvec3, 0}, {Utils::Type::mat4x3, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_float, 0}, {Utils::Type::dvec3, 0}, {Utils::Type::_float, 0}, {Utils::Type::_double, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::vec2, 0}, {Utils::Type::dvec3, 0}, {Utils::Type::_float, 0}, {Utils::Type::_double, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_double, 0}, {Utils::Type::_float, 0}, {Utils::Type::dvec2, 0}, {Utils::Type::vec3, 0} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::dmat3x4, 0}, {Utils::Type::_double, 0}, {Utils::Type::_float, 0}, {Utils::Type::dvec2, 0} }, is_nested_struct });
+
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::_float, 3}, {Utils::Type::dvec3, 0}, {Utils::Type::_double, 2} }, is_nested_struct });
+			m_test_cases.push_back(testCase{(Utils::Shader::STAGES) stage, { {Utils::Type::dmat2x3, 2}, {Utils::Type::mat2x3, 2}, {Utils::Type::dvec2, 0} }, is_nested_struct });
```

Comment:
Hmm, I faced with the following error [TestResults.qpa.txt](https://github.com/KhronosGroup/VK-GL-CTS/files/5918799/2333.-.TestResults.qpa.txt) trying this test.
```
struct TestStruct {
   dmat2x3 a[2];
   mat2x3 b[2];
   dvec2 c;
};
struct OuterStruct {
    TestStruct inner_struct_a;
    TestStruct inner_struct_b;
};
layout (location = 0, xfb_offset = 0) flat out OuterStruct goku;
```
It exceeds MAX_TRANSFORM_FEEDBACK_INTERLEAVED_COMPONENTS limit which is defined as 64 on Intel/Mesa3D. Nvidia (binary driver under Linux) passes it because Nvidia has higher limit defined as 128

Reply:
Indeed, I missed this case, thanks! I should have added an assert.

As a solution I propose to move this case outside of `for (bool is_nested_struct : {false, true})` and test it only without the nested struct.

Reply:
Now the case runs uses non-nested struct. Also added an assert to make sure the tests do not exceed the minimum value of MAX_TRANSFORM_FEEDBACK_INTERLEAVED_COMPONENTS.